locksmith/src/middlewares/signatureValidationMiddleware.ts

```diff
@@ -0,0 +1,70 @@
+import sigUtil from 'eth-sig-util'
+import ethJsUtil from 'ethereumjs-util'
+import Base64 from '../utils/base64'
+import { Request, Response } from 'express-serve-static-core'
+
+namespace SignatureValidationMiddleware {
+  function extractToken(req: Request) : String| null {
+    if (
+      req.headers.authorization &&
+      req.headers.authorization.split(' ')[0] === 'Bearer'
+    ) {
+      return req.headers.authorization.split(' ')[1]
+    } else {
+      return null
+    }
+  }
+
+  function validatePayload(payload: any, signee: string): Boolean{
```

Comment:
I think we need a type with lock as an optional sub-thing of payload

Reply:
This would be pretty easy to do.

```typescript
interface Payload {
  lock: Lock // NB. assumes that there is a type for Lock.
}
```

Since locks will probably be used and passed around a lot, it would be helpful to type them. And I think it should be pretty straightforward to do, no rabbit holes to fall into.

Reply:
So there is a longer story here, but the short version is that we have a representation of this in the creator dashboard https://github.com/unlock-protocol/unlock/blob/master/unlock-app/src/structured_data/unlockLock.js

While it is trivial to add it here, it will be pulled out in the subsequent code to follow this PR to unify what this code represents. If possible i'd like to do this only once

Reply:
That sounds reasonable and prudent to me